File: src/main/java/org/manapart/lumberjack/BaseBlock.java
package org.manapart.lumberjack;

public class BaseBlock {
    private final int worldX;
    private final int worldZ;
    private final int localX;
    private final int localZ;
    private final boolean isTree;
    private boolean isTrunk;
    private boolean isHarvestable;

    public BaseBlock(int localX, int localZ, int worldX, int worldZ, boolean isTree) {
        this.localX = localX;
        this.localZ = localZ;
        this.worldX = worldX;
        this.worldZ = worldZ;
        this.isTree = isTree;
    }

    public int getDistance(BaseBlock other) {
        return (int) Math.sqrt(Math.pow(worldX - other.worldX, 2) + Math.pow(worldZ - other.worldZ, 2));
    }

    public int getWorldX() {
        return worldX;
    }

    public int getWorldZ() {
        return worldZ;
    }

    public boolean isTree() {
        return isTree;
    }

    public boolean isTrunk() {
        return isTrunk;
    }

    public void setTrunk(boolean trunk) {
        isTrunk = trunk;
    }

    public boolean isHarvestable() {
        if (isTrunk) {
            return true;
        }
        return isHarvestable;
    }

    public void setHarvestable(boolean harvestable) {
        isHarvestable = harvestable;
    }

    public int getLocalZ() {
        return localZ;
    }

    public int getLocalX() {
        return localX;
    }
}

File: src/main/kotlin/org/manapart/lumberjack/BaseBlock.kt
package org.manapart.lumberjack

import kotlin.math.pow
import kotlin.math.sqrt

data class BaseBlock(val localX: Int, val localZ: Int, val worldX: Int, val worldZ: Int, val isTree: Boolean) {
    var isTrunk = false
    private var isHarvestable = false

    fun setHarvestable(harvestable: Boolean){
        this.isHarvestable = harvestable
    }

    fun shouldHarvest(): Boolean {
        return isTrunk || isHarvestable
    }

    fun getDistance(other: BaseBlock): Int {
        val xDist = (worldX - other.worldX).toDouble().pow(2.0)
        val yDist = (worldZ - other.worldZ).toDouble().pow(2.0)
        return sqrt(xDist + yDist).toInt()
    }
}
